python/GafferImageUI/ImageViewUI.py

```diff
@@ -1606,12 +1611,18 @@ def _update( self, *unused ) :
 			compareImage = _firstValidImagePlug( self.__nodeSet[-1] )
 
 		self.__node["compare"]["image"].setInput( compareImage )
-		self.__node["compare"]["catalogueOutput"].setValue( "output:%i" % self.__catalogueOutput if self.__catalogueOutput > 0 else "" )
+		if self.__catalogueOutput != "":
+			Gaffer.NodeAlgo.applyPreset( self.__node["compare"]["catalogueOutput"], self.__catalogueOutput )
+		else:
+			self.__node["compare"]["catalogueOutput"].setValue( "" )
 
 		# Icon
 
-		if self.__catalogueOutput > 0:
-			icon = "catalogueOutput%i.png" % self.__catalogueOutput
+		if self.__catalogueOutput != "":
+			if self.__catalogueOutput in [ "1", "2", "3", "4" ]:
+				icon = "catalogueOutput%s.png" % self.__catalogueOutput
```

Comment:
One option here would be to see if `catalogueOutput{foo}.png` exists, and then only fall back to `catalogueOutputHeader.png` if it doesn't. Might be a nice touch to allow people to provide an icon.

Reply:
We'd want to check that via `GafferUI.Image._qtPixmapFromFile()`, so that the query is cached and we don't hit the filesystem each time (bearing in mind the recent work we did to stop Qt stat()ing icons needlessly).

Reply:
Applied suggested changes.